WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductImageUploaderTests.swift

```diff
@@ -693,6 +731,154 @@ final class ProductImageUploaderTests: XCTestCase {
+        }
+    }
+
+    func test_activeUploads_are_tracked_through_storage_with_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+
+        _ = CurrentValueSubject<[ProductImageUploaderKey], Never>([])
+        let mockImageUploader = MockProductImageUploader()
+        mockImageUploader.activeUploadsKeys = []
+
+        var activeUploads: [ProductImageUploaderKey] = []
+        activeUploadsSubscription = mockImageUploader.activeUploads.sink { keys in
+            activeUploads = keys
+        }
+
+        let key = ProductImageUploaderKey(siteID: siteID,
+                                         productOrVariationID: productID,
+                                         isLocalID: false)
+
+        // When - simulate an uploading status
+        mockImageUploader.activeUploadsKeys = [key]
+
+        // Then
+        XCTAssertEqual(activeUploads.count, 1)
+        XCTAssertEqual(activeUploads.first?.siteID, key.siteID)
+        XCTAssertEqual(activeUploads.first?.productOrVariationID, key.productOrVariationID)
+        XCTAssertEqual(activeUploads.first?.isLocalID, key.isLocalID)
+
+        // When - simulate completion of upload
+        mockImageUploader.activeUploadsKeys = []
+
+        // Then
+        XCTAssertEqual(activeUploads.count, 0)
+    }
+
+    func test_background_upload_notice_is_sent_when_there_are_active_uploads_with_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+
+        let mockImageUploader = MockProductImageUploader()
+
+        let key = ProductImageUploaderKey(siteID: siteID,
+                                         productOrVariationID: productID,
+                                         isLocalID: false)
+
+        let noticePresenter = MockNoticePresenter()
+
+        // When - No active uploads
+        mockImageUploader.activeUploadsKeys = []
+        mockImageUploader.sendBackgroundUploadNoticeIfNeeded(key: key, using: noticePresenter)
+
+        // Then
+        XCTAssertTrue(mockImageUploader.sendBackgroundUploadNoticeIfNeededWasCalled)
+
+        // Reset flag for next check
+        mockImageUploader.sendBackgroundUploadNoticeIfNeededWasCalled = false
+
+        // When - With active uploads
+        mockImageUploader.activeUploadsKeys = [key]
+        mockImageUploader.sendBackgroundUploadNoticeIfNeeded(key: key, using: noticePresenter)
+
+        // Then
+        XCTAssertTrue(mockImageUploader.sendBackgroundUploadNoticeIfNeededWasCalled)
+    }
+
+    func test_reset_clears_storage_state_with_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+        let mockImageUploader = MockProductImageUploader()
+
+        // When
+        mockImageUploader.reset()
+
+        // Then
+        XCTAssertTrue(mockImageUploader.resetWasCalled, "Reset should be called")
```

Comment:
I don't understand why we are firing a method over `MockProductImageUploader` and checking that the method was called. We should instead be creating an instance of `ProductImageUploader` and testing that it clears the storage when `reset` method is called. 

Please correct me if I am missing something.